virtual_ecosystem/models/abiotic_simple/microclimate.py

```diff
@@ -101,62 +100,47 @@ def run_microclimate(
             data=data,
             reference_data=data[var + "_ref"].isel(time_index=time_index),
             leaf_area_index_sum=leaf_area_index_sum,
-            layer_roles=layer_roles,
+            layer_structure=layer_structure,
             layer_heights=data["layer_heights"],
             upper_bound=upper,
             lower_bound=lower,
             gradient=gradient,
         ).rename(var)
 
     # Mean atmospheric pressure profile, [kPa]
-    output["atmospheric_pressure"] = (
-        (data["atmospheric_pressure_ref"])
-        .isel(time_index=time_index)
-        .where(output["air_temperature"].coords["layer_roles"] != "soil")
-        .rename("atmospheric_pressure")
-        .T
+    # VIVI: the previous code implied this should only be filled for atmospheric roles,
+    # but that didn't happen.
+    output["atmospheric_pressure"] = layer_structure.from_template()
+    output["atmospheric_pressure"][:] = data["atmospheric_pressure_ref"].isel(
+        time_index=time_index
     )
 
```

Comment:
That was always a bit of a hack, I just filled all the non-soil layers with the reference data, even the canopy layers that are nan. So it would be ideal if we could assign the values only to the true above ground layers, do we have that somewhere in the layer structure? Or would it make sense for it to be where layer heights is not nan? Same for atmospheric CO2.

Reply:
Yeah - was thinking about this. Longer term, I suspect there are more elegant ways to handle the canopy layers, but for the moment I think it makes sense to provide a `LayerStructure.set_filled_canopy()` method, that the `plants` module and testing can use to add a new index layer. That could _technically_ be viewed as Data, but it makes sense to manage that specific bit of information in the context of the other layer indices. 

I was also thinking we could then support the generation of aggregate indices: `LayerStructure.get_indices(['AB', 'FC', 'SF'])` might return `[0,1,2,3,11]`.

Reply:
I think this would be useful and agree that for now we can just fill the whole array (above ground), that should not break anything. Do you want me to do that or are you already on it?